package dmGraphics;

message TextureImage
{
    enum TextureFormat
    {
        TEXTURE_FORMAT_LUMINANCE         = 0;
        TEXTURE_FORMAT_RGB               = 1;
        TEXTURE_FORMAT_RGBA              = 2;
        TEXTURE_FORMAT_RGB_DXT1          = 3;
        TEXTURE_FORMAT_RGBA_DXT1         = 4;
        TEXTURE_FORMAT_RGBA_DXT3         = 5;
        TEXTURE_FORMAT_RGBA_DXT5         = 6;
        TEXTURE_FORMAT_RGB_PVRTC_2BPPV1  = 7;
        TEXTURE_FORMAT_RGB_PVRTC_4BPPV1  = 8;
        TEXTURE_FORMAT_RGBA_PVRTC_2BPPV1 = 9;
        TEXTURE_FORMAT_RGBA_PVRTC_4BPPV1 = 10;
    }

    message Image
    {
        required uint32 width           = 1;
        required uint32 height          = 2;
        required TextureFormat format   = 3;
        repeated uint32 mip_map_offset  = 4;
        repeated uint32 mip_map_size    = 5;
        required bytes data             = 6;
    }

    repeated Image alternatives = 1;
}
